Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2003-02-14
Initial Package Version: 4.0.3
Origin: Shadow Mailing List
Description: Fixes PAM Compile Issues

diff -uNr shadow-4.0.3.orig/configure.in shadow-4.0.3/configure.in
--- shadow-4.0.3.orig/configure.in	2002-03-13 00:36:40.000000000 -0000
+++ shadow-4.0.3/configure.in	2003-06-26 21:59:00.000000000 -0000
@@ -265,10 +265,13 @@
 	AC_CHECK_LIB(pam, pam_start,
 		[AC_DEFINE(USE_PAM)
 		LIBPAM="-lpam"
+		SAVED_LIBS=$LIBS
+		LIBS="$LIBS $LIBPAM"
 		AC_CHECK_LIB(pam_misc, main,
 			[LIBPAM="$LIBPAM -lpam_misc"],
 			AC_MSG_ERROR(libpam_misc is missing)
 		)]
+		LIBS=$SAVED_LIBS
 		[AC_MSG_CHECKING(use login access checking if PAM not used)
 		AC_DEFINE(LOGIN_ACCESS)
 		AC_MSG_RESULT(yes)]
diff -Naur shadow-4.0.3.orig/libmisc/Makefile.am shadow-4.0.3/libmisc/Makefile.am
--- shadow-4.0.3.orig/libmisc/Makefile.am	2002-03-09 23:11:55.000000000 -0000
+++ shadow-4.0.3/libmisc/Makefile.am	2003-06-26 21:59:00.000000000 -0000
@@ -58,4 +58,4 @@
 	valid.c \
 	xmalloc.c
 
-libmisc_la_LIBADD = $(LIBCRACK)
+libmisc_la_LIBADD = $(LIBCRACK) $(LIBPAM)
